codegen/src/test/resources/software/amazon/awssdk/codegen/poet/rules/endpoint-resolve-interceptor.java

```diff
@@ -30,6 +31,11 @@ public SdkRequest modifyRequest(Context.ModifyRequest context, ExecutionAttribut
             .getAttribute(SdkInternalExecutionAttribute.ENDPOINT_PROVIDER);
         try {
             Endpoint result = provider.resolveEndpoint(ruleParams(context, executionAttributes)).join();
+            Optional<String> hostPrefix = hostPrefix(executionAttributes.getAttribute(SdkExecutionAttribute.OPERATION_NAME),
+                                                     context.request());
+            if (hostPrefix.isPresent() && !AwsEndpointProviderUtils.disableHostPrefixInjection(executionAttributes)) {
+                result = AwsEndpointProviderUtils.addHostPrefix(result, hostPrefix.get());
+            }
```

Comment:
Is it common enough that disableHostPrefixInjection will be true that we should put the calculation of hostPrefix within this if statement?

The compiler or runtime might be smart enough to do that automatically, but it also might not be...

Reply:
I'll see if I can shuffle things around a bit!